mathesar/rpc/explorations.py

```diff
@@ -50,6 +51,52 @@ def from_model(cls, model):
         )
 
 
+class ExplorationDef(TypedDict):
+    """
+    Definition about a runnable exploration.
+
+    Attributes:
+        base_table_oid: The OID of the base table of the exploration on the database.
+        initial_columns: A list describing the columns to be included in the exploration.
+        display_names: A map between the actual column names on the database and the alias to be displayed.
+        transformations: A list describing the transformations to be made on the included columns.
+        parameters: A dict describing the properties to be applied while retrieving records e.g. limit, offset, filter, order_by, etc.
```

Comment:
I'd really prefer to have the `parameters` dict exploded into keyword arguments. I.e., the exploration definition would have `limit`, `offset`, `filter`, `order_by` attributes.

This will make validation easier and more natural. It also makes the documentation provided by this class more useful.